app/views/mediatorConfigModal.html

```diff
@@ -38,11 +38,26 @@ <h4 class="modal-title" id="addMediatorConfigLabel">
           </div>
 
           <div ng-if="def.type === 'number'">
-            <input class="form-control" type="number" ng-model="mediator.config[def.param]" ng-disabled="update">
+            <input class="form-control" type="number" ng-model="mediator.config[def.param]">
           </div>
 
           <div ng-if="def.type === 'option'">
-            <select class="form-control" ng-options="val for val in def.values" ng-model="mediator.config[def.param]">
+            <select class="form-control" ng-options="val for val in def.values" ng-model="mediator.config[def.param]"/>
+          </div>
+
+          <div ng-if="def.type === 'map'">
+            <table style="width: 100%;">
+              <tr ng-repeat="(mapping, _v) in mediator.config[def.param]">
+                <td>{{mapping}}</td>
+                <td><input class="form-control" type="text" ng-model="mediator.config[def.param][mapping]"></td>
+                <td><button class="btn btn-danger btn-xs" ng-click="removeMapping(def.param, mapping)"><i class="glyphicon glyphicon-remove"></td>
+              </tr>
+              <tr>
+                <td><input class="form-control" type="text" ng-model="newMapping.key"></td>
+                <td><input class="form-control" type="text" ng-model="newMapping.value"></td>
+                <td><button class="btn btn-primary btn-xs" ng-click="addNewMapping(def.param)" ng-disabled="!newMapping.key"><i class="glyphicon glyphicon-plus"></td>
```

Comment:
You may want to disable this if the key already exists in the map and provide some sort of indication of this.